scripts/packager-codes/si-packagers-refresh.py

```diff
@@ -0,0 +1,214 @@
+#   https://www.gov.si/assets/organi-v-sestavi/UVHVVR/Varna-hrana/Odobritev-obrata/Obrati-Zivila-O_ang.pdf
+# - download last version of https://github.com/tabulapdf/tabula-java/releases
+# - convert the pdf file into csv (update release number and file name):
+# $ java -jar tabula-1.0.5-jar-with-dependencies.jar   Obrati-Zivila-O_ang.pdf \
+#     --lattice --format CSV --pages all > slovenian_packaging_raw.csv
+
+# RUN
+update: api_key
+python3 si-packagers-refresh.py
+
+# POSTPROCESSING
+- deactivate the virtual environment:
+deactivate
+'''
+
+import polars as pl
+import re
+import requests
+import sys
+from time import sleep
+
+def clean_code(input_code: str) -> str:
+	# remove double spaces
+	input_code = input_code.replace('  ', ' ')
+
+	# SI H-1015 ES
+	if input_code.endswith('ES'):
+		input_code = input_code.replace('ES', '').strip()
+
+	# SI M-1035 SI
+	if input_code.endswith('SI'):
+		input_code = input_code.replace(' SI', '').strip()
+
+	# SI H-731, SI 731
+	if ',' in input_code:
+		input_code = "".join(input_code.split(', ')[1])
+
+	# SI H - 728, SI H 728, SI H-728, also with M
+	input_code = input_code.replace('H - ', 'H-')
+	input_code = input_code.replace('H ', 'H-')
+	input_code = input_code.replace('M - ', 'M-')
+	input_code = input_code.replace('M ', 'M-')
+
+	# SI - 907 -> SI 907
+	input_code = input_code.replace(' - ', ' ')
+	input_code = input_code.replace(' -', ' ')
+	input_code = input_code.replace('SI-', 'SI ')
+
+	# SI1194
+	if 'SI ' not in input_code:
+		input_code = input_code.replace('SI', 'SI ')
+	# SI M1106
+	if 'M-' not in input_code:
+		input_code = input_code.replace('M', 'M-')
+
+	return input_code
+
+
+def clean_address(input_address: str) -> str:
+	# special character because 
+	# sometimes new line between 2 addreses
+	# sometimes line for single address split in 2
+	input_address = "<>".join(input_address.split('\r'))
+
+	# fetch last occurence
+	# words 123A, place, 4567 city name
+	# Á found in a city name (PROSENJAKOVCI -PÁRTOSFALVA)
+	pattern = r'(([a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.]+\d+[ABCDEFGIJ]?),(?:[a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.\<\>]+,\s*)?[\<\>]*(\s*\d{4}[a-zčćžđšA-ZČĆŽĐŠŽÁ\s\-\.\<\>]+)$)'
+
+	match = re.search(pattern, input_address)
+
+	
+	if match:
+		output_address = (f"{match.group(2).strip().title()}, {match.group(3).replace('<>', ' ').strip().title()}")
+	else:
+		# MOŠNJE , MOŠNJE, 4240 RADOVLJICA -> no street number (also DIJAŠKA ULICA , 5220 TOLMIN)
+		# instead, fetch "something, postal_code city"
+		pattern_2 = r'(([a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.]+),(\s*\d{4})([a-zčćžđšA-ZČĆŽĐŠŽÁ\s\-\.\<\>]+)$)'
+		match_2 = re.search(pattern_2, input_address)
+		if match_2:
+			output_address = (f"{match_2.group(2).strip().title()}, {match_2.group(3).replace('<>', ' ').strip().title()}")		
+		else:
+			print("Match problem", input_address)
+			output_address = input_address
+
+	return output_address
+
+
+def convert_address_to_lat_lng(address_to_convert: str) -> str:
+	# free plan: 1 request per second
+	sleep(1)
```

Comment:
You are patient, with 1s per request, I would have made a [dbm cache](https://docs.python.org/3/library/dbm.html#module-dbm.gnu) to avoid issuing same requests again ;-)

Reply:
I applied suggestions

For the fun. There are only 11 addresses repeated. It spared 11 seconds. But this is cool, and can be used for other countries woth bigger files.